ratis-server/src/main/java/org/apache/ratis/server/raftlog/LogProtoUtils.java

```diff
@@ -137,8 +137,9 @@ public static LogEntryProto removeStateMachineData(LogEntryProto entry) {
   }
 
   private static LogEntryProto replaceStateMachineDataWithSerializedSize(LogEntryProto entry) {
-    return replaceStateMachineEntry(entry,
+    LogEntryProto replaced = replaceStateMachineEntry(entry,
         StateMachineEntryProto.newBuilder().setLogEntryProtoSerializedSize(entry.getSerializedSize()));
+    return copy(replaced);
```

Comment:
@duongkame I thought `replaced` is already a newly build object on heap and not dependent on `entry` anymore, so copy might be redundant here?

Reply:
@symious , `replaced` is built by replacing some parts (i.e. `StateMachineEntryProto`) of the `LogEntryProto` .  The remaining parts need to be copied.

Reply:
@szetszwo Thank you for the explaination, I misunderstood that `LogEntryProto.newBuilder(proto)` will perform a deep copy.